### PR TITLE
Convert Nordbits to AngleSharp

### PR DESCRIPTION
**Pull request by ngosang**
Untested #7292

@garfield69 please, test this one. I think we are removing the first row/torrent. Could you check?

### CODE REVIEW

File: src/Jackett.Common/Indexers/Norbits.cs [hunk 1]
```diff
@@ -630,12 +607,11 @@ private void LatencyNow()
...
-        /// <returns>JQuery Object</returns>
-        private CQ FindTorrentRows()
+        /// <returns>List of rows</returns>
+        private IHtmlCollection<IElement> FindTorrentRows(IHtmlDocument dom)
         {
             // Return all occurencis of torrents found
-            //return _fDom["#content > table > tr"];
-            return _fDom["# torrentTable > tbody > tr:not(:first)"];
+            return dom.QuerySelectorAll("#torrentTable > tbody > tr:not(:first)");
         }
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/Norbits.cs]: As a 1 line function, maybe this should be inlined?

**Comment by cadatoiva** [comment]
It would appear that either :not() or :first is not supported by CSS Selectors. This should fix the problem. Please test again

**Comment by cadatoiva** [comment]
No, that's fine. This shows that the change worked, but I have to figure out how to Cast it correctly back to the intended type.

File: src/Jackett.Common/Indexers/Norbits.cs [hunk 2]
```diff
@@ -630,13 +607,9 @@ private void LatencyNow()
...
-        /// <returns>JQuery Object</returns>
-        private CQ FindTorrentRows()
-        {
-            // Return all occurencis of torrents found
-            //return _fDom["#content > table > tr"];
-            return _fDom["# torrentTable > tbody > tr:not(:first)"];
-        }
+        /// <returns>List of rows</returns>
+        private IHtmlCollection<IElement> FindTorrentRows(IHtmlDocument dom) =>
+            (IHtmlCollection<IElement>)dom.QuerySelectorAll("#torrentTable > tbody > tr").Skip(1);
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/Norbits.cs]:
```suggestion
           dom.QuerySelectorAll("#torrentTable > tbody > tr").Skip(1).ToCollection();
```

**Comment by cadatoiva** [comment]
@garfield69 That should do it, try once more.

File: src/Jackett.Common/Indexers/Norbits.cs [hunk 3]
```diff
@@ -398,19 +375,19 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                             MinimumSeedTime = 172800 // 48 hours
                         };
 
-                        var genres = tRow.Find("span.genres").Text();
+                        var genres = row.QuerySelector("span.genres").TextContent;
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/Norbits.cs]: Progress, but no banana quite yet.

some rows do not contain any genres.
line 378 is generating
```
2020-03-01 18:02:03 | Error | Jackett.Common.IndexerException: Exception (norbits): Object reference not set to an instance of an object.  ---> System.Exception: Parse error  ---> System.NullReferenceException: Object reference not set to an instance of an object.    at Jackett.Common.Indexers.Norbits.PerformQuery(TorznabQuery query) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\Norbits.cs:line 378    --- End of inner exception stack trace ---    at Jackett.Common.Indexers.BaseWebIndexer.OnParseError(String results, Exception ex) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\BaseIndexer.cs:line 830    at Jackett.Common.Indexers.Norbits.PerformQuery(TorznabQuery query) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\Norbits.cs:line 402    at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\BaseIndexer.cs:line 323    --- End of inner exception stack trace ---    at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\BaseIndexer.cs:line 343    at Jackett.Common.Indexers.BaseWebIndexer.ResultsForQuery(TorznabQuery query) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\BaseIndexer.cs:line 803    at Jackett.Server.Controllers.ResultsController.Results(ApiSearch requestt) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Server\Controllers\ResultsController.cs:line 224
-- | -- | --
2020-03-01 18:02:02 | Error | Error on 20200301180202 for Norbits.txtSystem.NullReferenceException: Object reference not set to an instance of an object.    at Jackett.Common.Indexers.Norbits.PerformQuery(TorznabQuery query) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\Norbits.cs:line 378     Error, unable to parse result     at Jackett.Common.Indexers.Norbits.PerformQuery(TorznabQuery query) in C:\Users\Garfield69\Documents\GitHub\Jackett\src\Jackett.Common\Indexers\Norbits.cs:line 378
```
I think you first have to `var genres = row.QuerySelector("span.genres")` then check !null and then use `.TextContent` ?

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/Norbits.cs]:
```suggestion
                        var genres = row.QuerySelector("span.genres")?.TextContent;
```

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/Norbits.cs]: That checks for null, and the if on the next line shows that this one should have been checking for null in the first place.

**Comment by garfield69** [comment]
found a similar issue with the imdb at 383, where most rows do not have one. I've applied the `?` and now we are getting results! \o/

**Pull request merged**